core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -502,6 +512,7 @@ public void setAuthenticator(Authenticator authenticator) {
 
     public void setBinBindVariable(int index, long address, int valueLen) throws SqlException {
         bindVariableService.setBin(index, this.binarySequenceParamsPool.next().of(address, valueLen));
+        freezeRecvBuffer = true;
```

Comment:
Looks like we have a similar problem in `setDateBindVariable`: it also uses a flyweight view over the receive buffer which may lead to input corruption. It might be a good idea to change that method too, i.e. via a copying char sink.

Reply:
I changed `setDateBindVariable` to create a copy of a string similar to what `setStrBindVariable` does